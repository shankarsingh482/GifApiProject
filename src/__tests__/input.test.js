import React from 'react';
import {shallow} from 'enzyme';
import Input from '../components/inputComponent';


import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Input',()=>
{
    it('Input For Search Component should be defined',()=>{
        expect(Input).toBeDefined();
    })
    it('Input component should render Correctly ',()=>
    {
        const tree= shallow(
                    <Input name='input test'/>
        );
        expect(tree).toMatchSnapshot();
    })
});