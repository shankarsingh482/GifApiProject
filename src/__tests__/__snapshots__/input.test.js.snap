// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input component should render Correctly  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    name="input test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "onChange": undefined,
      "placeholder": undefined,
      "type": undefined,
      "value": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onChange": undefined,
        "placeholder": undefined,
        "type": undefined,
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
